openshift-knative-operator/pkg/serving/extension.go

```diff
@@ -88,6 +89,17 @@ func (e *extension) Reconcile(ctx context.Context, comp v1alpha1.KComponent) err
 		}
 	}
 
+	// Disable metrics backend by default due to OOM issues, for more check SRVKS-679
+	// Until now by default nothing was set and knative/pkg set the backend to `prometheus`
+	// As an exception if the user sets the backend explicitly in the CR the value is not modified.
+	if observCMData, ok := ks.Spec.CommonSpec.Config[observabilityCMName]; ok {
```

Comment:
I don't think the nesting here is necessary. Reading from `nil` maps is fine AFAIK.

As such `if _, ok := ks.Spec.CommonSpec.Config[observabilityCMName]["metrics.backend-destination"]; !ok {`

should be fine. A test would be great (there should be unit tests for this function anyway)